### PR TITLE
Multiplicative dynamic cost model

### PR DESCRIPTION
**Pull request by rvbansal**
Changes the dynamic cost model to make multiplicative changes, rather than additive changes, to the cost factor.

### CODE REVIEW

File: src/CostModelDynamicLevel.sol [hunk 1]
```diff
@@ -230,19 +230,21 @@ contract CostModelDynamicLevel is ICostModel {
...
+        costFactorAtFullUtilization,
+        currentCostFactorInOptimalZone_.mulWadUp(FixedPointMathLib.WAD + optimalZoneRate * deviation_)
       );
     } else {
-      // Cost factor decreases with `timeDelta` and `utilization - uOpt`, but with a floor set at
-      // `costFactorAtZeroUtilization`.
-      uint256 delta_ = optimalZoneRate.mulWadUp((uOpt - utilization_) * timeDelta_);
-      if (delta_ > currentCostFactorInOptimalZone_ - costFactorAtZeroUtilization) return costFactorAtZeroUtilization;
-      else return currentCostFactorInOptimalZone_ - delta_;
+      uint256 deviation_ = timeDelta_.mulDivUp(uOpt - utilization_, FixedPointMathLib.WAD);
```
Comment (chad-js) [inline src/CostModelDynamicLevel.sol:241]: Maybe better to do the div by wad on the next line for precision?

Comment (rvbansal) [reply src/CostModelDynamicLevel.sol]: Makes sense -- Changed, so I multiply everything together first and then scale by WAD^2.

74e479a8887741792f407c6d1fc6fe0814ffaa01

Comment (chad-js) [reply src/CostModelDynamicLevel.sol]: Screen Shot 2023-09-11 at 11 42 57 AM

I'm still confused why we need to do WAD ** 2 in the numerator of the new calc? 😅

Comment (rvbansal) [reply src/CostModelDynamicLevel.sol]: Ahh sorry -- so for the case where `utilization > uOpt`, we want to grow the cost factor as `currentCostFactor*(1 + growthFactor)`. Basically we need the WAD^2, so the underlying calculation is `currentCostFactor*(WAD^2 + growthFactor*WAD^2) / WAD^2`. 

To be clearer, I'll change this so it matches what we do in the `utilization < uOpt` case, where we just compute the delta and then add it.

Comment (chad-js) [reply src/CostModelDynamicLevel.sol]: ohhhh I see. thanks for explaining

Comment (rvbansal) [reply src/CostModelDynamicLevel.sol]: Changed here: 1334ec044cba05685106225f06384f18a0e7101f

**Review by chad-js — APPROVED** [review]

**Pull request merged**